homeassistant/components/mqtt/models.py

```diff
@@ -48,6 +53,35 @@ class ReceiveMessage:
 MessageCallbackType = Callable[[ReceiveMessage], None]
 
 
+class SubscriptionDebugInfo(TypedDict):
+    """Class for holding subscription debug info."""
+
+    messages: deque[ReceiveMessage]
+    count: int
+
+
+class EntityDebugInfo(TypedDict):
+    """Class for holding entity based debug info."""
+
+    subscriptions: dict[str, SubscriptionDebugInfo]
+    discovery_data: DiscoveryInfoType
+    transmitted: dict[str, dict[str, deque[TimestampedPublishMessage]]]
+
+
+class TriggerDebugInfo(TypedDict):
+    """Class for holding trigger based debug info."""
+
+    device_id: str
+    discovery_data: DiscoveryInfoType
+
+
+class DebugInfo(TypedDict, total=False):
```

Comment:
@cdce8p I have used a TypedDict structure and moved hass.data[MQTT_DATA_DEBUG_INFO] to the new dataclass structure. Is this the way to go?

Reply:
Basically yes. Why set `total=False`? AFAICT it's initialized with both keys present. If the keys aren't required, something like this wouldn't be safe
```py
mqtt_data.debug_info["entities"]
```